jax/image/scale.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import enum
+import math
+from typing import Callable, Sequence, Tuple, Union
+
+from jax import lax
+from jax import numpy as jnp
+import numpy as np
+
+
+def _lanczos_kernel(radius: float):
+  def kernel(x: np.ndarray) -> np.ndarray:
```

Comment:
Just interested, why is this numpy and not jnp?

Reply:
The span calculation is a function of the scale and the input shapes, and both are fixed at compile time in the current implementation. So we can use classic NumPy to calculate the weights; no need to lift it into JAX and, say, run it on the GPU.